### PR TITLE
Adding in Muon Efficiency Maps

### PR DESCRIPTION
**Pull request by ocolegro**
Adding Muon Efficiency maps to the DQM GUI for fast tracking efficiency studies.

### CODE REVIEW

**Comment by jfernan2** [comment]
please test

File: DQM/TrackingMonitor/src/TrackEfficiencyMonitor.cc [hunk 1]
```diff
@@ -282,8 +303,22 @@ void TrackEfficiencyMonitor::analyze(const edm::Event& iEvent, const edm::EventS
   iSetup.get<IdealMagneticFieldRecord>().get(bField); 
   iSetup.get<TrackerRecoGeometryRecord>().get(theTracker);
   theNavigation = new DirectTrackerNavigation(theTracker);
-  
-  
+
+  edm::Handle<edm::View<reco::Muon> > muons; 
+  iEvent.getByToken(muonToken_,muons);
```
Comment (fioriNTU) [inline DQM/TrackingMonitor/src/TrackEfficiencyMonitor.cc]: @ocolegro please add a check on the validity of the muon collection here, this part may cause issues in the Online clients if the muons are not reconstructed

**Comment by jandrea** [comment]
I am trying to understand the origin of the error. It doesn't seem related to the modifications made by Owen (?). 

----- Begin Fatal Exception 19-Jun-2018 06:45:06 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=MixingModule label='mix'
Exception Message:
RootEmbeddedFileSequence no input files specified for secondary input source.
----- End Fatal Exception -------------------------------------------------

**Comment by jfernan2** [comment]
please test

**Comment by fabiocos** [comment]
@jfernan2 @dmitrijus is this PR ok to move forward?

File: DQM/TrackingMonitor/src/TrackEfficiencyMonitor.cc (hunk 1 above)
Comment (jfernan2) [reply DQM/TrackingMonitor/src/TrackEfficiencyMonitor.cc]: I agree on the comment by @fioriNTU A prevention is needed please

**Comment by dmitrijus** [comment]
+1

**Comment by fioriNTU** [comment]
also this comment should be addressed:
https://github.com/cms-sw/cmssw/pull/23614#discussion-diff-196314737R308
Currently we are not protected against crashes.

**Comment by dmitrijus** [comment]
-1
Waiting for the author's feedback.

**Comment by ocolegro** [comment]
Muon collection checking was added.

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
@ocolegro the warning compiling is still there, can you please fix it?

**Comment by jfernan2** [comment]
please test
________________________________________

**Comment by jfernan2** [comment]
@ocolegro sorry but I am afraid this pR is adding a lot of lines to the logs like:

AlCaReco/SiStrip/MechanicalView/TEC/PLUS/wheel_9/backward_petals/petal_6/ring_5/mono_modules/module_470439598:ClusterDigiPosition__det__4704395

in https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23614/28982/runTheMatrix-results/136.85_RunEGamma2018A+RunEGamma2018A+HLTDR2_2018+RECODR2_2018reHLT_skimEGamma_Prompt_L1TEgDQM+HARVEST2018_L1TEgDQM/step3_RunEGamma2018A+RunEGamma2018A+HLTDR2_2018+RECODR2_2018reHLT_skimEGamma_Prompt_L1TEgDQM+HARVEST2018_L1TEgDQM.log

You should only enable real warning IMHO

**Comment by fabiocos** [comment]
please test

**Comment by jfernan2** [comment]
@ocolegro You are adding thousands of lines at the end of the log files like this:
AlCaReco/SiStrip/MechanicalView/TEC/MINUS/wheel_1:ClusterPosition__TEC__MINUS__wheel__1,NumberOfClusterPerRingVsTrendVar,NumberOfClusterTrend__TEC__MINUS__wheel__1
which are not real warnings but root file directory structure.
See e.g. https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23614/29270/runTheMatrix-results/136.85_RunEGamma2018A+RunEGamma2018A+HLTDR2_2018+RECODR2_2018reHLT_skimEGamma_Prompt_L1TEgDQM+HARVEST2018_L1TEgDQM/step3_RunEGamma2018A+RunEGamma2018A+HLTDR2_2018+RECODR2_2018reHLT_skimEGamma_Prompt_L1TEgDQM+HARVEST2018_L1TEgDQM.log

**Comment by andrius-k** [comment]
+1

**Comment by kpedro88** [comment]
+1

**Comment by VinInn** [comment]
It seems to me that this PR is polluting the top level tracking folder
https://goo.gl/qCkciN
please FIX
-1

**Comment by VinInn** [comment]
@mtosi @JanFSchulte

**Comment by mtosi** [comment]
@ocolegro @jandrea are you planning to fix this ? or shall I do it ? thanks

**Comment by mtosi** [comment]
in addition, may I ask why you are not using the DQMGenericClient for doing the efficiency plots ?

**Pull request merged**